shared-bindings/microcontroller/Pin.h

```diff
@@ -37,5 +37,7 @@ void assert_pin(mp_obj_t obj, bool none_ok);
 void assert_pin_free(const mcu_pin_obj_t* pin);
 
 bool common_hal_mcu_pin_is_free(const mcu_pin_obj_t* pin);
+void common_hal_never_reset_pin(const mcu_pin_obj_t* pin);
+void common_hal_reset_pin(const mcu_pin_obj_t* pin);
```

Comment:
I know this hasn't been said anywhere but I only use the `common_hal_` prefix for shared API that maps to Python API. For cross-port functions, they should just be added here without a prefix like `assert_pin` and `assert_pin_free` already.

Reply:
That makes total sense, but now I'm not really sure what to name these. I was reluctant to redo the original functions so I wouldn't risk breaking functionality in other sections of the ports and shared bindings, so I mimicked what was done with `common_hal_mcu_pin_is_free`, which is a common-hal alias on top of a port-level function. 

I think there's a distinction between functions and structures that ONLY exist at the port level, which are essentially helpers, and stuff that's actually exposed in shared bindings, even if it isn't a 1:1 map to python. Do you think a new prefix would be appropriate for functions like this? `port_hal` or some such?